src/Rector/Rector/MethodCall/ReplaceCommandArgsIoWithPropertiesRector.php

```diff
@@ -0,0 +1,126 @@
+<?php
+declare(strict_types=1);
+
+namespace Cake\Upgrade\Rector\Rector\MethodCall;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Param;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PHPStan\Type\ObjectType;
+use Rector\PhpParser\Node\BetterNodeFinder;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+final class ReplaceCommandArgsIoWithPropertiesRector extends AbstractRector
+{
+    public function __construct(
+        protected BetterNodeFinder $betterNodeFinder,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace `$args` and `$io` parameters in Command classes with `$this->args` and `$this->io`',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class TestCommand extends Command
+{
+    public function execute(Arguments $args, ConsoleIo $io)
+    {
+        $io->out('Hello');
+        $this->someMethod($args, $io);
+    }
+
+    protected function someMethod(Arguments $args, ConsoleIo $io): void
+    {
+        $someArg = $args->getArgument('some');
+        $io->warning('Warn');
+    }
+}
+CODE_SAMPLE,
+                    <<<'CODE_SAMPLE'
+class TestCommand extends Command
+{
+    public function execute()
+    {
+        $this->io->out('Hello');
+        $this->someMethod();
+    }
+
+    protected function someMethod(): void
+    {
+        $someArg = $this->args->getArgument('some');
+        $this->io->warning('Warn');
+    }
+}
+CODE_SAMPLE,
+                ),
+            ],
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof ClassMethod) {
+            return null;
+        }
```

Comment:
The operating spacing here doesn't seem to be consistent with the cakephp cs rules. Perhaps we aren't running those phpcs rules on this repo?